Release v5.0.0 (#37)
ReachFive/identity-android-sdk: CHANGELOG.md

# Changelog

## v5.0.0 (01/08/2019)

### Breaking changes

- The default scopes for the `signup` and `loginWithPassword` methods used to be `openid`, `email`, `phone` and `profile`.

> They are now the allowed scopes set up in your client's configuration through the console.

Reviewer: ```suggestion
The new default is to use the scopes defined for your client via the ReachFive console. Note that you must initialise the client through the `initialize` method for the scopes to be set, or an empty value will be used.
```

- All the data models used for requests were moved in a sub-folder named `requests` (`com.reach5.identity.sdk.core.models` -> `com.reach5.identity.sdk.core.models.requests`).
- The profile's data passed as an argument to the `signup` method is no longer a `Profile` but a [`ProfileSignupRequest`](https://developer.reach5.co/api/identity-android/#profile-signup-request).

> Thus `ProfileSignupRequest` provides the same attributes than `Profile` without the identifiers and authentication details and with the `password` field.

Reviewer: ```suggestion
Note that `ProfileSignupRequest` contains the same attributes as `Profile` but minus identifiers and authentication details. This model naturally contains a `password` field.
```

> - The `User` data model was renamed [`OpenIdUser`](https://developer.reach5.co/api/identity-android/#openid-user).

Reviewer: ```suggestion
- The `User` data model was renamed to [`OpenIdUser`](https://developer.reach5.co/api/identity-android/#openid-user).
```

> - The `AuthToken.idToken` field became optional since it's not returned if the `openid` scope is not provided.

Reviewer: ```suggestion
- The `AuthToken.idToken` field became optional because it isn't returned when the `openid` scope is not provided.
```

- The `openid` scope is no longer provided by default to the `loginWithProvider` method through the `WebViewProvider`.

> - The `logout` method can now also be used to disconnect sessions created with a provider.

Reviewer: ```suggestion
- The `logout` method now disconnects all sessions including those created with a provider.
```
